sys/include/flash_utils.h

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (C) 2022 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    sys_flash_utils Utility functions, macros, and types for
+ *                              read-only memory
+ * @ingroup     sys
+ *
+ * This modules adds utility functions, macros, and functions for read-only
+ * memory. The goal is to hide the differences between modified architectures
+ * that map flash into the data address space (e.g. ARM) and those which
+ * doesn't (e.g. most AVR, Xtensa).
+ *
+ * # Usage
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{.c}
+ * #include "flash_utils.h"
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * No module needs to be selected, this is a header-only implementation that
+ * is always available.
+ *
+ * # Porting Code to Use `flash_utils`
+ *
+ * This is mainly targeting applications developers to ease developing apps
+ * that work well on both legacy modified Harvard architectures (e.g. ATmega)
+ * and modern modified Harvard architectures (e.g. ARM, ATtiny, ...) as well
+ * as von-Neumann machines.
+ *
+ * The intention is to limit in-tree use to small small number of modules that
```

Comment:
Is this word duplication intentionally?

```suggestion
 * The intention is to limit in-tree use to a very small number of modules that
```